cpp/tests/unary/unary_ops_test.cu

```diff
@@ -55,6 +55,155 @@ cudf::test::fixed_width_column_wrapper<T> create_expected_columns(cudf::size_typ
     }
 }
 
+template <typename T>
+struct cudf_logical_test : public cudf::test::BaseFixture {};
+
+TYPED_TEST_CASE(cudf_logical_test, cudf::test::NumericTypes);
+
+TYPED_TEST(cudf_logical_test, LogicalNot)
+{
+    cudf::size_type colSize = 1000;
+    std::vector<TypeParam> h_input_v(colSize, false);
+    std::vector<cudf::experimental::bool8> h_expect_v(colSize);
+
+    std::transform(
+        std::cbegin(h_input_v),
+        std::cend(h_input_v),
+        std::begin(h_expect_v),
+        [] (TypeParam e) -> cudf::bool8 {
+            return static_cast<cudf::bool8>(!e);
+        });
+
+    cudf::test::fixed_width_column_wrapper<TypeParam>                 input    (std::cbegin(h_input_v),  std::cend(h_input_v));
+    cudf::test::fixed_width_column_wrapper<cudf::experimental::bool8> expected (std::cbegin(h_expect_v), std::cend(h_expect_v));
+
+    auto output = cudf::experimental::unary_operation(input, cudf::experimental::unary_op::NOT);
+
+    cudf::test::expect_columns_equal(expected, output->view());
+}
+
+TYPED_TEST(cudf_logical_test, SimpleLogicalNot)
+{
+    cudf::test::fixed_width_column_wrapper<TypeParam>                 input    {{ true,  true,  true,  true  }};
+    cudf::test::fixed_width_column_wrapper<cudf::experimental::bool8> expected {{ false, false, false, false }};
+    auto output = cudf::experimental::unary_operation(input, cudf::experimental::unary_op::NOT);
+    cudf::test::expect_columns_equal(expected, output->view());
+}
+
+template <typename T>
+struct cudf_math_test : public cudf::test::BaseFixture {};
+
+TYPED_TEST_CASE(cudf_math_test, cudf::test::NumericTypes);
+
+TYPED_TEST(cudf_math_test, ABS)
+{
+    using T = TypeParam;
+
+    cudf::size_type colSize = 1000;
+    std::vector<T> h_input_v(colSize);
+    std::vector<T> h_expect_v(colSize);
+
+    std::iota(
+        std::begin(h_input_v),
+        std::end(h_input_v),
+        -1 * colSize);
+
+    std::transform(
+        std::cbegin(h_input_v),
+        std::cend(h_input_v),
+        std::begin(h_expect_v),
+        [] (auto e) { return std::abs(e); });
+
+    cudf::test::fixed_width_column_wrapper<T> input    (std::cbegin(h_input_v),  std::cend(h_input_v));
+    cudf::test::fixed_width_column_wrapper<T> expected (std::cbegin(h_expect_v), std::cend(h_expect_v));
+
+    auto output = cudf::experimental::unary_operation(input, cudf::experimental::unary_op::ABS);
+
+    cudf::test::expect_columns_equal(expected, output->view());
+}
+
+TYPED_TEST(cudf_math_test, SQRT)
+{
+    using T = TypeParam;
+
+    cudf::size_type colSize = 175;
```

Comment:
Please use a size that spans multiple thread blocks.

Reply:
Done.